subsys/net/lib/nrf_cloud/coap/src/nrf_cloud_coap.c

```diff
@@ -638,7 +640,27 @@ int nrf_cloud_coap_shadow_state_update(const char * const shadow_json)
 		return -EACCES;
 	}
 
-	err = nrf_cloud_coap_patch(COAP_SHDW_RSC, NULL, (uint8_t *)shadow_json,
+	err = nrf_cloud_coap_patch(COAP_SHDW_REP_RSC, NULL, (uint8_t *)shadow_json,
+				   strlen(shadow_json),
+				   COAP_CONTENT_FORMAT_APP_JSON, true, NULL, NULL);
+	if (err < 0) {
+		LOG_ERR("Failed to send PATCH request: %d", err);
+	} else if (err > 0) {
+		LOG_RESULT_CODE_ERR("Error from server:", err);
+	}
+	return err;
+}
+
+int nrf_cloud_coap_shadow_desired_update(const char * const shadow_json)
```

Comment:
consider adding something like:
`static int shadow_update(const char * const resource, const char * const shadow_json)`
to avoid some duplicate code.